inspirehep/modules/forms/validators/simple_fields.py

```diff
@@ -0,0 +1,186 @@
+
+
+def isbn_syntax_validation(form, field):
+    """Validate ISBN syntax."""
+    message = "The provided ISBN is invalid - it should look \
+                similar to '1413304540', '1-4133-0454-0', '978-1413304541' or \
+                '978-1-4133-0454-1'."
+
+    if field.data and not is_isbn(field.data):
+        raise StopValidation(message)
+
+
+def does_exist_in_inspirehep(query, collection=None):
+    """Check if there exist an item in the db which satisfies query.
+
+    :param query: http query to check
+    :param collection: collection to search in; by default searches in
+        the default collection
+    """
+    import requests
+    from json import loads
+
+    params = {
+        'p': query,
+        'of': 'id'
+    }
+
+    if collection:
+        params['cc'] = collection
+
+    json_reply = requests.get(
+        "http://inspirehep.net/search?", params=params).text
+
+    reply = loads(json_reply)
+    # not an array -> invalid answer
+    if not isinstance(reply, list):
+        raise ValueError('Invalid server answer.')
+
+    # non-empty answer -> duplicate
+    if len(reply):
+        return True
+
+    return False
+
+
+def inspirehep_duplicated_validator(inspire_query, property_name):
+    """Check if a record with the same doi already exists.
+
+    Needs to be wrapped in a function with proper validator signature.
+    """
+    if does_exist_in_inspirehep(inspire_query):
+        url = "http://inspirehep.net/search?" + urlencode({'p': inspire_query})
+        raise ValidationError(
+            'There exists already an item with the same %s. '
+            '<a target="_blank" href="%s">See the record.</a>'
+            % (property_name, url)
+        )
+
+
+def duplicated_doi_validator(form, field):
+    """Check if a record with the same doi already exists."""
+    doi = field.data
+    # TODO: local check for duplicates
+    if not doi:
+        return
+    if current_app.config.get('PRODUCTION_MODE'):
+        inspirehep_duplicated_validator('doi:' + doi, 'DOI')
+
+
+def duplicated_arxiv_id_validator(form, field):
+    """Check if a record with the same arXiv ID already exists."""
+    arxiv_id = field.data
+    # TODO: local check for duplicates
+    if not arxiv_id:
+        return
+    if current_app.config.get('PRODUCTION_MODE'):
+        inspirehep_duplicated_validator(
+            '035__a:oai:arXiv.org:' + arxiv_id, 'arXiv ID')
+
+
+def pdf_validator(form, field):
+    """Validate that url points to PDF."""
+    import requests
+
+    def get_content_type(url):
+        session = requests.Session()
+        try:
+            response = session.head(
+                url,
+                allow_redirects=True
+            )
+        except:
+            return
+        return response.headers['content-type']
+
+    message = "Please, provide an accessible direct link to a PDF document."
+
+    if field.data and get_content_type(field.data) != 'application/pdf':
+        raise StopValidation(message)
```

Comment:
True that it might take some seconds, but at least it would not allow the form to submit.
This is currently the main pain point of the workflow, where users post URLs that are either invalid, or we are not allowed to fetch.